### ISSUE TITLE
Decorator for extraction options from strategy file

### ISSUE THREAD

**Issue by tjgalvin** [tjgalvin/flint#179]
When using a strategy file the general usage is something like:

```
# Get the options out of the file
wsclean_options = get_options_from_strategy(
    strategy=strategy, mode="wsclean", round=current_round
)
# Pass in the options to the wsclean function, note the update_wsclean_options
wsclean_cmds = task_wsclean_imager.map(
    in_ms=cal_mss,
    wsclean_container=field_options.wsclean_container,
    update_wsclean_options=unmapped(wsclean_options),
    fits_mask=fits_beam_masks,
)
```

It might be nice to add a decorator to the `task_wsclean_imager` function

```
@wrapper_options_from_strategy
def task_wsclean_imager(...):
```
so that we can do something like
```
wsclean_cmds = task_wsclean_imager.map(
    in_ms=cal_mss,
    wsclean_container=field_options.wsclean_container,
    update_wsclean_options=unmapped(wsclean_options),
    fits_mask=fits_beam_masks,
    strategy=strategy, 
    mode="wsclean", 
    round=current_round
)
```

The problem is that should this be used we would need to either:
1 - ensure that every such function we wrap has a consistent `update_options` parameter
2 - update the `wrapper_options_from_strategy` to accept the keyword to update. Something like:

```
def wrapper_options_from_strategy(keyword):
   def wrapper(fn):
        def _get_options_from_strategy(strategy, *args, mode="wsclean", round="iinitial", **kwargs):
            options = get_options_from_strategy(strategy, round, mode)
            kwargs[keyword] = options
            return fn(*args, **kwargs)
       return _get_options_from_strategy
   return wrapper
```

so that the keyword to update is a necessary parameter to supply when decorating

```
@wraooer_options_from_strategy(keyword="update_wsclean_options")
def task_wsclean_imager(...):
```

**Comment by tjgalvin** [comment]
I am playing around with this under the `stratdec` branch. 

Personally, I prefer option 2. I think being explicit here is perhaps better then being sneaky. It also means we can further specify the `mode` and `operations` aspects of the keywords.

**Comment by tjgalvin** [comment]
Being consider in #180 . Seems to be working well.

**Comment by tjgalvin** [comment]
Closed in #180
